onnxruntime/core/providers/cuda/tensor/scatter_elements_impl.cu

```diff
@@ -7,6 +7,33 @@
 namespace onnxruntime {
 namespace cuda {
 
+template <typename T, typename Tin, bool OUTERAXIS>
+__global__ void _ScatterElementsKernel2D(
+    const int max_dim,  // max dim on the scattered axis
+    const T* input_data,
+    const Tin* indices_data,
+    const int64_t indices_size,
+    const fast_divmod indices_stride_row,
+    const T* updates,
+    const int64_t output_row_size,
+    T* output_data) {
+  CALCULATE_ELEMENTWISE_INDEX_OR_EXIT(indices_index, indices_size);
+
+  int row, col, data_idx;
+  indices_stride_row.divmod(indices_index, row, col);
+  int dim = (int)(indices_data[indices_index]);
```

Comment:
is there potential overflow here?

Reply:
Agree. In general those use fast_divmod do not support large size. This is common bad in many current implementation. Yet we may force use gsl:narrow_cast check later for all possible space.